config/v1/types_node.go

```diff
@@ -0,0 +1,156 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Node holds cluster-wide information about node specific features.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Node struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec NodeSpec `json:"spec"`
+
+	// status holds observed values.
+	// +optional
+	Status NodeStatus `json:"status"`
+}
+
+type NodeSpec struct {
+	// CgroupMode determines the cgroups version on the node
+	// +optional
+	CgroupMode CgroupMode `json:"cgroupMode,omitempty"`
+
+	// WorkerLatencyProfile determins the how fast the kubelet is updating
+	// the status and corresponding reaction of the cluster
+	// +optional
+	WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+}
+
+type NodeStatus struct {
+	// WorkerLatencyProfileStatus provides the current status of WorkerLatencyProfile
+	// +optional
+	WorkerLatencyProfileStatus WorkerLatencyProfileStatus `json:"workerLatencyProfileStatus,omitempty"`
+}
+
+type CgroupMode string
+
+const (
+	CgroupMode_Empty   = "" // Empty string will always use the Default
+	CgroupMode_v1      = "v1"
+	CgroupMode_v2      = "v2"
+	CgroupMode_Default = CgroupMode_v1
```

Comment:
Don't use `_` in names, we rarely do that, instead use `CGroupV1`, `CGroupV2`. I'd suggest dropping `CgroupMode_Default` entirely, same with `CgroupMode_empty` both are expressed as empty/not set. No need to add these as constants. Just be explicit that when not set default v1 is used on the field itself.

Reply:
@rphillips for the comment on cgroups

Reply:
This naming is carried over from the enhancement... The naming should stay the same.

Reply:
Fine with the naming, but the part about empty vs default still applies, both mean the exact same thing so I'd only add v1 and v2.

Reply:
@deads2k  mentioned adding support for empty and default.

Reply:
Ack

Reply:
There are still 3 nits to get it merged:
1. `CgroupModeEmpty` instead of `CgroupMode_Empty` to follow all the other types we have, ie. don't use `_` in the name
2. you need to explicitly tell that `CgroupModeEmpty` is of `CgroupMode`  iow. `CgroupModeEmpty CgroupMode = ""` w/o it's a string and not the type you wanted it to be
3. rename `CgroupMode` to `CgroupModeType` to follow other types like this.

@rphillips I was told to let you know about it :smile:

Reply:
Thanks... +1